components/IngredientDetails.js

```diff
@@ -1,12 +1,27 @@
 import { useRouter } from "next/router";
-import ingredients from "@/assets/ingredients.json";
+import { useEffect, useState } from "react";
+import ingredientsData from "@/assets/ingredients.json";
 import IngredientItem from "@/components/IngredientItem";
 import { Container, BackLink } from "@/_styles";
 
 const IngredientDetails = () => {
   const router = useRouter();
   const { id } = router.query;
-  const ingredient = ingredients.find((ingredient) => ingredient._id === id);
+  const [ingredient, setIngredient] = useState(null);
+
+  useEffect(() => {
+    if (id) {
+      const allIngredients = ingredientsData;
+      const foundIngredient = allIngredients.find(
+        (ingredient) => ingredient._id === id
+      );
+      setIngredient(foundIngredient);
+    }
+  }, [id]);
```

Comment:
There is no need for a useEffect or useState here. The way you had it prior to this PR was already ideal:
```suggestion
 const ingredient = ingredients.find((ingredient) => ingredientsData._id === id);
```